Create global variables from PDB

b1llow opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

...

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

...

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

closes #3625

b1llow committed:
Add RzBuf API

XVilka commented on librz/bin/pdb/omap.c:
```diff
@@ -88,21 +71,19 @@ static int binary_search(unsigned int *A, int key, int imin, int imax) {
  * \return int
  */
-RZ_API int rz_bin_pdb_omap_remap(RZ_NONNULL RzPdbOmapStream *omap_stream, int address) {
+static ut64 rz_bin_pdb_omap_remap(RZ_NONNULL RzPdbOmapStream *omap_stream, ut64 address) {
```
If it's static, please remove `rz_` prefix

XVilka commented on librz/bin/pdb/omap.c:
```diff
@@ -133,3 +109,17 @@ RZ_API int rz_bin_pdb_omap_remap(RZ_NONNULL RzPdbOmapStream *omap_stream, int ad
 	return omap_entry->to + (address - omap_entry->from);
 }
+
+RZ_API ut64 rz_bin_pdb_to_rva(RZ_NONNULL const RzPdb *pdb, const PDBSectionOffset *section_offset) {
```
Please add Doxygen

XVilka commented on librz/bin/pdb/tpi.c:
```diff
@@ -403,34 +444,34 @@ RZ_API bool rz_bin_pdb_type_is_fwdref(RZ_NONNULL RzPdbTpiType *t) {
  * \param t RzPdbTpiType
  * \return RzList *
```
Update Doxygen too

XVilka commented on librz/include/rz_util/rz_buf.h:
```diff
@@ -349,6 +350,18 @@ DEFINE_RZ_BUF_WRITE_OFFSET_BLE(64)
 #undef DEFINE_RZ_BUF_READ_OFFSET_BLE
 #undef DEFINE_RZ_BUF_WRITE_OFFSET_BLE
 
+static inline ut8 rz_buf_peek(RZ_NONNULL RzBuffer *b) {
```
Please add Doxygen for those

XVilka commented on test/db/formats/pdb:
```diff
@@ -1,3 +1,40 @@
+idp bins/pdb/Project1.pdb
+avg
+EOF
+EXPECT=<<EOF
```
Could you also please add a test that checks if those (just a few, at least two) global variables are referenced? E.g. using both `avgx` and `pd`/`pdf` commands?

XVilka replied:
This way, we make sure that these globals are picked up by the analysis.

XVilka reviewed (changes requested):
rz-bindgen failure:
```
Run python3 rz-bindgen/src/lint.py \
Type comment at <rizin/librz/bin/pdb/tpi.h:543:24> lacks angle brackets
Error: Process completed with exit code 1.
```

I think it might make sense to try fixing also https://github.com/rizinorg/rizin/issues/955 in the same PR (if it's easy enough, of course)

wargio commented on librz/bin/pdb/tpi.h:
```diff
@@ -487,6 +534,22 @@ typedef struct {
+	TpiCVType complete_class; // class/structure that owns the vftable
+	TpiCVType override_vftable; // vftable from which this vftable is derived
+	ut32 vfptr_offset; // offset of the vfptr to this table, relative to the start of the object layout.
+	RzPVector /* <char *> */ method_names; // array of names.
```
```suggestion
	RzPVector /*<char *>*/ method_names; // array of names.
```

b1llow committed:
Improve pdb parse
- Set global variables
- Use RzPVector instead of RzList in Tpi_LF_FieldList->substructs
- Add some symbol parse
- Skip all unknown leaf type
